{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "coordinated-adams",
   "metadata": {},
   "outputs": [],
   "source": [
    "def reference_genome():\n",
    "    #Coronavirus_2_isolate_genome.fasta\n",
    "    #Set script to look in reference_genome folder. \n",
    "\n",
    "    from pathlib import Path\n",
    "\n",
    "    \n",
    "    path=Path(\"reference_genome/\")\n",
    "    file_name=''\n",
    "    for entry in path.iterdir():\n",
    "        print(\"*\"*100)\n",
    "        print(\"\\nReference Genome Used:\\n '\" + entry.name+ \"'\\n\" )\n",
    "        file_name=entry.name\n",
    "        break\n",
    "        \n",
    "\n",
    "    \n",
    "    #Open File\n",
    "    fh=open(\"reference_genome/\"+file_name, 'r')\n",
    "    fn=fh.readlines()\n",
    "    corona_genome=''\n",
    "    for line in fn:\n",
    "        if line.startswith(\">\"):\n",
    "            continue\n",
    "        else:\n",
    "            corona_genome+=line.strip()\n",
    "    return corona_genome\n",
    "    \n",
    "def read_bed_file():\n",
    "    import pandas as pd\n",
    "    from pathlib import Path\n",
    "    file_list=[]\n",
    "    path=Path(\"bed/\")\n",
    "    file_name=''\n",
    "\n",
    "    for entry in path.iterdir():\n",
    "        \n",
    "        file_name=entry.name\n",
    "        save_file= file_name.strip(\".bed\")\n",
    "        bed_file=open(\"bed/\"+file_name, \"r\")\n",
    "        \n",
    "        #On MacOS filters the .DS_Store file. \n",
    "        if file_name.startswith(\".\"):\n",
    "            continue\n",
    "        else: \n",
    "            \n",
    "            bed_file_list=bed_file.readlines()\n",
    "\n",
    "            yield bed_file_list,save_file\n",
    "        \n",
    "        \n",
    "        #print(\"bed file found :'\" + entry.name+\"'\" )\n",
    "        #file_list.append(entry.name)\n",
    "\n",
    "def create_dataframe(a_list,corona_genome):\n",
    "    \n",
    "    import pandas as pd\n",
    "    from pathlib import Path\n",
    "    root=\"output_vcf/\"\n",
    "    new_file=[]\n",
    "    print(\"*\"*100)\n",
    "    print(\"Converting File: \"+ a_list[1]+\".bed\\n\")\n",
    "    print(\"Indels found: \")\n",
    "    path=Path(\"bed/\") \n",
    "    for line in a_list[0]:\n",
    "\n",
    "        df_list=line.split(\"\\t\")\n",
    "        ID=df_list[len(df_list)-1].strip()\n",
    "        deletion_start=int(df_list[len(df_list)-3].strip())-1\n",
    "        deletion_end=int(df_list[len(df_list)-2].strip())\n",
    "        #print(deletion_start,deletion_end)\n",
    "        if ID.startswith(\"d\") or ID.startswith(\"i\"):\n",
    "\n",
    "\n",
    "            #Sets pos equal to pos-1\n",
    "            #to follow the sample vcf file. \n",
    "            b=line.split()\n",
    "            b[2]=b[1]\n",
    "            b='\\t'.join(b)\n",
    "            #print(b)\n",
    "\n",
    "\n",
    "            original=b.strip()\n",
    "\n",
    "\n",
    "            original+=\"\\t\" +corona_genome[deletion_start:deletion_end]+\"\\t\"+corona_genome[deletion_start]\n",
    "\n",
    "            new_file.append(original)\n",
    "\n",
    "\n",
    "\n",
    "            print(\" •\",ID[len(ID)-len(ID):] ,\": \",corona_genome[deletion_start:deletion_end])\n",
    "\n",
    "\n",
    "        else:    \n",
    "            original=line.strip()\n",
    "            original+=\"\\t\" +ID[len(ID)-len(ID)]+\"\\t\"+ID[len(ID)-1]\n",
    "            new_file.append(original)\n",
    "\n",
    "        \n",
    "    df = pd.DataFrame([sub.split(\"\\t\") for sub in new_file],index=None,columns=[\"#CHROM\", \"POS-1\", \"POS\", 'ID',\"REF\", \"ALT\"])\n",
    "    df.drop('POS-1', inplace=True, axis=1)\n",
    "    \n",
    "    df.to_csv(root+a_list[1]+\".vcf\",sep=\"\\t\",header=True, index=False)    \n",
    "    print(\"\\nPreview:  \" +a_list[1]+ \".vcf\\n\\n\",df,\"\\n\")\n",
    "\n",
    "\n",
    "    \n",
    "\n",
    "\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "excellent-google",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "****************************************************************************************************\n",
      "\n",
      "Reference Genome Used:\n",
      " 'Coronavirus_2_isolate_genome.fasta'\n",
      "\n",
      "Converting Files: \n",
      "P.1.bed\n",
      "B_1_351.bed\n",
      "P_1.bed\n",
      "B_1_429.bed\n",
      "B.1.429.bed\n",
      "****************************************************************************************************\n",
      "Converting File: P.1.bed\n",
      "\n",
      "Indels found: \n",
      " • del_11288 :  GTCTGGTTTT\n",
      " • ins_28263 :  GA\n",
      "\n",
      "Preview:  P.1.vcf\n",
      "\n",
      "          #CHROM    POS         ID         REF ALT\n",
      "0   NC_045512v2    241      C241T           C   T\n",
      "1   NC_045512v2    733      T733C           T   C\n",
      "2   NC_045512v2   2749     C2749T           C   T\n",
      "3   NC_045512v2   3037     C3037T           C   T\n",
      "4   NC_045512v2   3828     C3828T           C   T\n",
      "5   NC_045512v2   5648     A5648C           A   C\n",
      "6   NC_045512v2   6319     A6319G           A   G\n",
      "7   NC_045512v2   6613     A6613G           A   G\n",
      "8   NC_045512v2  11287  del_11288  GTCTGGTTTT   G\n",
      "9   NC_045512v2  12778    C12778T           C   T\n",
      "10  NC_045512v2  13860    C13860T           C   T\n",
      "11  NC_045512v2  14408    C14408T           C   T\n",
      "12  NC_045512v2  17259    G17259T           G   T\n",
      "13  NC_045512v2  21614    C21614T           C   T\n",
      "14  NC_045512v2  21621    C21621A           C   A\n",
      "15  NC_045512v2  21638    C21638T           C   T\n",
      "16  NC_045512v2  21974    G21974T           G   T\n",
      "17  NC_045512v2  22132    G22132T           G   T\n",
      "18  NC_045512v2  22812    A22812C           A   C\n",
      "19  NC_045512v2  23012    G23012A           G   A\n",
      "20  NC_045512v2  23063    A23063T           A   T\n",
      "21  NC_045512v2  23403    A23403G           A   G\n",
      "22  NC_045512v2  23525    C23525T           C   T\n",
      "23  NC_045512v2  24642    C24642T           C   T\n",
      "24  NC_045512v2  25088    G25088T           G   T\n",
      "25  NC_045512v2  26149    T26149C           T   C\n",
      "26  NC_045512v2  28167    G28167A           G   A\n",
      "27  NC_045512v2  28262  ins_28263          GA   G\n",
      "28  NC_045512v2  28512    C28512G           C   G\n",
      "29  NC_045512v2  28877    A28877T           A   T\n",
      "30  NC_045512v2  28878    G28878C           G   C\n",
      "31  NC_045512v2  28881    G28881A           G   A\n",
      "32  NC_045512v2  28882    G28882A           G   A\n",
      "33  NC_045512v2  28883    G28883C           G   C \n",
      "\n",
      "****************************************************************************************************\n",
      "Converting File: B_1_351.bed\n",
      "\n",
      "Indels found: \n",
      " • del_11288 :  GTCTGGTTTT\n",
      " • del_22281 :  ACTTTACTTG\n",
      "\n",
      "Preview:  B_1_351.vcf\n",
      "\n",
      "          #CHROM    POS         ID         REF ALT\n",
      "0   NC_045512v2    174      G174T           G   T\n",
      "1   NC_045512v2    241      C241T           C   T\n",
      "2   NC_045512v2   1059     C1059T           C   T\n",
      "3   NC_045512v2   3037     C3037T           C   T\n",
      "4   NC_045512v2   5230     G5230T           G   T\n",
      "5   NC_045512v2  10323    A10323G           A   G\n",
      "6   NC_045512v2  11287  del_11288  GTCTGGTTTT   G\n",
      "7   NC_045512v2  14408    C14408T           C   T\n",
      "8   NC_045512v2  21801    A21801C           A   C\n",
      "9   NC_045512v2  22280  del_22281  ACTTTACTTG   A\n",
      "10  NC_045512v2  22813    G22813T           G   T\n",
      "11  NC_045512v2  23012    G23012A           G   A\n",
      "12  NC_045512v2  23063    A23063T           A   T\n",
      "13  NC_045512v2  23403    A23403G           A   G\n",
      "14  NC_045512v2  23664    C23664T           C   T\n",
      "15  NC_045512v2  25563    G25563T           G   T\n",
      "16  NC_045512v2  26456    C26456T           C   T\n",
      "17  NC_045512v2  28253    C28253T           C   T\n",
      "18  NC_045512v2  28887    C28887T           C   T \n",
      "\n",
      "****************************************************************************************************\n",
      "Converting File: P_1.bed\n",
      "\n",
      "Indels found: \n",
      " • del_11288 :  GTCTGGTTTT\n",
      " • ins_28263 :  GA\n",
      "\n",
      "Preview:  P_1.vcf\n",
      "\n",
      "          #CHROM    POS         ID         REF ALT\n",
      "0   NC_045512v2    241      C241T           C   T\n",
      "1   NC_045512v2    733      T733C           T   C\n",
      "2   NC_045512v2   2749     C2749T           C   T\n",
      "3   NC_045512v2   3037     C3037T           C   T\n",
      "4   NC_045512v2   3828     C3828T           C   T\n",
      "5   NC_045512v2   5648     A5648C           A   C\n",
      "6   NC_045512v2   6319     A6319G           A   G\n",
      "7   NC_045512v2   6613     A6613G           A   G\n",
      "8   NC_045512v2  11287  del_11288  GTCTGGTTTT   G\n",
      "9   NC_045512v2  12778    C12778T           C   T\n",
      "10  NC_045512v2  13860    C13860T           C   T\n",
      "11  NC_045512v2  14408    C14408T           C   T\n",
      "12  NC_045512v2  17259    G17259T           G   T\n",
      "13  NC_045512v2  21614    C21614T           C   T\n",
      "14  NC_045512v2  21621    C21621A           C   A\n",
      "15  NC_045512v2  21638    C21638T           C   T\n",
      "16  NC_045512v2  21974    G21974T           G   T\n",
      "17  NC_045512v2  22132    G22132T           G   T\n",
      "18  NC_045512v2  22812    A22812C           A   C\n",
      "19  NC_045512v2  23012    G23012A           G   A\n",
      "20  NC_045512v2  23063    A23063T           A   T\n",
      "21  NC_045512v2  23403    A23403G           A   G\n",
      "22  NC_045512v2  23525    C23525T           C   T\n",
      "23  NC_045512v2  24642    C24642T           C   T\n",
      "24  NC_045512v2  25088    G25088T           G   T\n",
      "25  NC_045512v2  26149    T26149C           T   C\n",
      "26  NC_045512v2  28167    G28167A           G   A\n",
      "27  NC_045512v2  28262  ins_28263          GA   G\n",
      "28  NC_045512v2  28512    C28512G           C   G\n",
      "29  NC_045512v2  28877    A28877T           A   T\n",
      "30  NC_045512v2  28878    G28878C           G   C\n",
      "31  NC_045512v2  28881    G28881A           G   A\n",
      "32  NC_045512v2  28882    G28882A           G   A\n",
      "33  NC_045512v2  28883    G28883C           G   C \n",
      "\n",
      "****************************************************************************************************\n",
      "Converting File: B_1_429.bed\n",
      "\n",
      "Indels found: \n",
      "\n",
      "Preview:  B_1_429.vcf\n",
      "\n",
      "          #CHROM    POS       ID REF ALT\n",
      "0   NC_045512v2    241    C241T   C   T\n",
      "1   NC_045512v2   1059   C1059T   C   T\n",
      "2   NC_045512v2   3037   C3037T   C   T\n",
      "3   NC_045512v2  14408  C14408T   C   T\n",
      "4   NC_045512v2  17014  G17014T   G   T\n",
      "5   NC_045512v2  21600  G21600T   G   T\n",
      "6   NC_045512v2  22018  G22018T   G   T\n",
      "7   NC_045512v2  22917  T22917G   T   G\n",
      "8   NC_045512v2  23403  A23403G   A   G\n",
      "9   NC_045512v2  25563  G25563T   G   T\n",
      "10  NC_045512v2  26681  C26681T   C   T\n",
      "11  NC_045512v2  28272  A28272T   A   T\n",
      "12  NC_045512v2  28887  C28887T   C   T \n",
      "\n",
      "****************************************************************************************************\n",
      "Converting File: B.1.429.bed\n",
      "\n",
      "Indels found: \n",
      "\n",
      "Preview:  B.1.429.vcf\n",
      "\n",
      "          #CHROM    POS       ID REF ALT\n",
      "0   NC_045512v2    241    C241T   C   T\n",
      "1   NC_045512v2   1059   C1059T   C   T\n",
      "2   NC_045512v2   3037   C3037T   C   T\n",
      "3   NC_045512v2  14408  C14408T   C   T\n",
      "4   NC_045512v2  17014  G17014T   G   T\n",
      "5   NC_045512v2  21600  G21600T   G   T\n",
      "6   NC_045512v2  22018  G22018T   G   T\n",
      "7   NC_045512v2  22917  T22917G   T   G\n",
      "8   NC_045512v2  23403  A23403G   A   G\n",
      "9   NC_045512v2  25563  G25563T   G   T\n",
      "10  NC_045512v2  26681  C26681T   C   T\n",
      "11  NC_045512v2  28272  A28272T   A   T\n",
      "12  NC_045512v2  28887  C28887T   C   T \n",
      "\n",
      ".vcf files saved in '/output_vcf' directory.\n"
     ]
    }
   ],
   "source": [
    "def main():\n",
    "    import pandas as pd\n",
    "    from pathlib import Path\n",
    "    reference=reference_genome()\n",
    "    \n",
    "    print(\"Converting Files: \")\n",
    "    path=Path(\"bed/\")\n",
    "    \n",
    "    #print files info\n",
    "    for entry in path.iterdir():\n",
    "        if entry.name.startswith(\".\"):\n",
    "            continue\n",
    "        else:\n",
    "            print(entry.name)\n",
    "        \n",
    "    \n",
    "    \n",
    "    \n",
    "    #Use Generator\n",
    "    for i in read_bed_file():\n",
    "        create_dataframe(i,reference)\n",
    "    \n",
    "    print(\".vcf files saved in '/output_vcf' directory.\")\n",
    "if __name__ == \"__main__\":\n",
    "    main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "realistic-truck",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"GTCTGGTTTT\" == \"GTCTGGTTTT\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "prescription-launch",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\r\n",
      "##nanopolish_window=MN908947.3:1-29902\r\n",
      "##artic-tools_version=0.2.6\r\n",
      "##INFO=<ID=TotalReads,Number=1,Type=Integer,Description=\"The number of event-space reads used to call the variant\">\r\n",
      "##INFO=<ID=SupportFraction,Number=1,Type=Float,Description=\"The fraction of event-space reads that support the variant\">\r\n",
      "##INFO=<ID=SupportFractionByStrand,Number=2,Type=Float,Description=\"Fraction of event-space reads that support the variant for each strand\">\r\n",
      "##INFO=<ID=BaseCalledReadsWithVariant,Number=1,Type=Integer,Description=\"The number of base-space reads that support the variant\">\r\n",
      "##INFO=<ID=BaseCalledFraction,Number=1,Type=Float,Description=\"The fraction of base-space reads that support the variant\">\r\n",
      "##INFO=<ID=AlleleCount,Number=1,Type=Integer,Description=\"The inferred number of copies of the allele\">\r\n",
      "##INFO=<ID=StrandSupport,Number=4,Type=Integer,Description=\"Number of reads supporting the REF and ALT allele, by strand\">\r\n",
      "##INFO=<ID=StrandFisherTest,Number=1,Type=Integer,Description=\"Strand bias fisher test\">\r\n",
      "##INFO=<ID=SOR,Number=1,Type=Float,Description=\"StrandOddsRatio test from GATK\">\r\n",
      "##INFO=<ID=RefContext,Number=1,Type=String,Description=\"The reference sequence context surrounding the variant call\">\r\n",
      "##INFO=<ID=Pool,Number=1,Type=String,Description=\"The pool name\">\r\n",
      "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\r\n",
      "##FILTER=<ID=PASS,Description=\"All filters passed\">\r\n",
      "##contig=<ID=MN908947.3>\r\n",
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tsample\r\n",
      "MN908947.3\t241\t.\tC\tT\t11724.9\tPASS\tTotalReads=1596;SupportFraction=0.937635;SupportFractionByStrand=0.985732,0.903369;BaseCalledReadsWithVariant=1403;BaseCalledFraction=0.874143;AlleleCount=1;StrandSupport=9,90,655,842;StrandFisherTest=127;SOR=0.0216397;RefContext=GGTTTCGTCCG;Pool=1\tGT\t1\r\n",
      "MN908947.3\t259\t.\tA\tG\t11146.2\tPASS\tTotalReads=1595;SupportFraction=0.955882;SupportFractionByStrand=0.93943,0.967585;BaseCalledReadsWithVariant=1234;BaseCalledFraction=0.769327;AlleleCount=1;StrandSupport=40,30,623,902;StrandFisherTest=21;SOR=0.981809;RefContext=CCGAAAGGTAA;Pool=1\tGT\t1\r\n",
      "MN908947.3\t1059\t.\tC\tT\t9113\tPASS\tTotalReads=1280;SupportFraction=0.940064;SupportFractionByStrand=0.956416,0.928948;BaseCalledReadsWithVariant=1243;BaseCalledFraction=0.96133;AlleleCount=1;StrandSupport=23,54,495,708;StrandFisherTest=14;SOR=0.316815;RefContext=TGACACCTTCA;Pool=2\tGT\t1\r\n",
      "MN908947.3\t3037\t.\tC\tT\t3175.6\tPASS\tTotalReads=394;SupportFraction=0.972971;SupportFractionByStrand=0.993188,0.936696;BaseCalledReadsWithVariant=317;BaseCalledFraction=0.800505;AlleleCount=1;StrandSupport=2,9,251,132;StrandFisherTest=28;SOR=1.29997;RefContext=TCTTTCTACCC;Pool=2\tGT\t1\r\n",
      "MN908947.3\t8083\t.\tG\tA\t7392.1\tPASS\tTotalReads=1008;SupportFraction=0.960753;SupportFractionByStrand=0.938385,0.983032;BaseCalledReadsWithVariant=837;BaseCalledFraction=0.823009;AlleleCount=1;StrandSupport=31,9,472,496;StrandFisherTest=37;SOR=0.178476;RefContext=CCAATGGAAAA;Pool=1\tGT\t1\r\n",
      "MN908947.3\t10319\t.\tC\tT\t9620.6\tPASS\tTotalReads=1580;SupportFraction=0.811754;SupportFractionByStrand=0.988627,0.677771;BaseCalledReadsWithVariant=1399;BaseCalledFraction=0.877666;AlleleCount=1;StrandSupport=8,290,673,609;StrandFisherTest=694;SOR=0.200004;RefContext=TTAAGCTTAAG;Pool=2\tGT\t1\r\n",
      "MN908947.3\t10344\t.\tA\tG\t15546\tPASS\tTotalReads=1576;SupportFraction=0.943195;SupportFractionByStrand=0.938795,0.946509;BaseCalledReadsWithVariant=1314;BaseCalledFraction=0.826935;AlleleCount=1;StrandSupport=41,48,636,851;StrandFisherTest=2;SOR=0.849502;RefContext=TCCTAAGACAC;Pool=2\tGT\t1\r\n",
      "MN908947.3\t14408\t.\tC\tT\t6843.7\tPASS\tTotalReads=894;SupportFraction=0.910932;SupportFractionByStrand=0.962575,0.852649;BaseCalledReadsWithVariant=760;BaseCalledFraction=0.844444;AlleleCount=1;StrandSupport=18,62,456,358;StrandFisherTest=83;SOR=0.536554;RefContext=CCCACCTACAA;Pool=2\tGT\t1\r\n",
      "MN908947.3\t14805\t.\tC\tT\t4985.1\tPASS\tTotalReads=800;SupportFraction=0.920278;SupportFractionByStrand=0.963937,0.86443;BaseCalledReadsWithVariant=768;BaseCalledFraction=0.946979;AlleleCount=1;StrandSupport=16,48,433,303;StrandFisherTest=63;SOR=0.768253;RefContext=GACTACTATCG;Pool=1\tGT\t1\r\n",
      "MN908947.3\t18424\t.\tA\tG\t8455.6\tPASS\tTotalReads=894;SupportFraction=0.959652;SupportFractionByStrand=0.973176,0.947614;BaseCalledReadsWithVariant=846;BaseCalledFraction=0.941046;AlleleCount=1;StrandSupport=11,25,410,448;StrandFisherTest=11;SOR=0.240631;RefContext=CACCTAATAAT;Pool=1\tGT\t1\r\n",
      "MN908947.3\t20056\t.\tG\tA\t579.3\tPASS\tTotalReads=65;SupportFraction=0.948464;SupportFractionByStrand=0.914277,0.999744;BaseCalledReadsWithVariant=61;BaseCalledFraction=0.924242;AlleleCount=1;StrandSupport=3,0,36,26;StrandFisherTest=6;SOR=0.0944619;RefContext=CAGAAGGTAGT;Pool=2\tGT\t1\r\n",
      "MN908947.3\t21008\t.\tC\tT\t5362.8\tPASS\tTotalReads=720;SupportFraction=0.923942;SupportFractionByStrand=0.969774,0.850645;BaseCalledReadsWithVariant=688;BaseCalledFraction=0.943759;AlleleCount=1;StrandSupport=13,41,430,236;StrandFisherTest=81;SOR=1.23166;RefContext=TGCAACTGTAC;Pool=1\tGT\t1\r\n",
      "MN908947.3\t21304\t.\tC\tT\t229.4\tPASS\tTotalReads=40;SupportFraction=0.926247;SupportFractionByStrand=0.931286,0.920678;BaseCalledReadsWithVariant=39;BaseCalledFraction=0.975;AlleleCount=1;StrandSupport=1,2,20,17;StrandFisherTest=0;SOR=0.781709;RefContext=AACCACGCGAA;Pool=2\tGT\t1\r\n",
      "MN908947.3\t23403\t.\tA\tG\t14778.4\tPASS\tTotalReads=1761;SupportFraction=0.964961;SupportFractionByStrand=0.939667,0.992442;BaseCalledReadsWithVariant=1434;BaseCalledFraction=0.805618;AlleleCount=1;StrandSupport=55,6,862,838;StrandFisherTest=99;SOR=0.0179994;RefContext=TCAGGATGTTA;Pool=1\tGT\t1\r\n",
      "MN908947.3\t25563\t.\tG\tT\t14239.4\tPASS\tTotalReads=1414;SupportFraction=0.982863;SupportFractionByStrand=0.981689,0.983714;BaseCalledReadsWithVariant=1219;BaseCalledFraction=0.859056;AlleleCount=1;StrandSupport=11,13,583,807;StrandFisherTest=0;SOR=0.836686;RefContext=TTTCAGAGCGC;Pool=2\tGT\t1\r\n",
      "MN908947.3\t25907\t.\tG\tT\t485.6\tPASS\tTotalReads=50;SupportFraction=0.959271;SupportFractionByStrand=0.954748,0.96086;BaseCalledReadsWithVariant=47;BaseCalledFraction=0.903846;AlleleCount=1;StrandSupport=1,1,12,36;StrandFisherTest=0;SOR=1.41737;RefContext=TTCAGGTGATG;Pool=1\tGT\t1\r\n",
      "MN908947.3\t27964\t.\tC\tT\t6973.4\tPASS\tTotalReads=733;SupportFraction=0.972892;SupportFractionByStrand=0.972142,0.973264;BaseCalledReadsWithVariant=653;BaseCalledFraction=0.881242;AlleleCount=1;StrandSupport=7,13,236,477;StrandFisherTest=0;SOR=0.802961;RefContext=ACAGTCATGTA;Pool=2\tGT\t1\r\n",
      "MN908947.3\t28472\t.\tC\tT\t5856.8\tPASS\tTotalReads=736;SupportFraction=0.936743;SupportFractionByStrand=0.958052,0.870921;BaseCalledReadsWithVariant=659;BaseCalledFraction=0.892954;AlleleCount=1;StrandSupport=23,23,533,157;StrandFisherTest=39;SOR=2.51441;RefContext=AATTCCCTCGA;Pool=2\tGT\t1\r\n",
      "MN908947.3\t28869\t.\tC\tT\t836.8\tPASS\tTotalReads=127;SupportFraction=0.865441;SupportFractionByStrand=0.990093,0.797034;BaseCalledReadsWithVariant=104;BaseCalledFraction=0.806202;AlleleCount=1;StrandSupport=0,17,45,65;StrandFisherTest=32;SOR=0.0141481;RefContext=AACTCCAGGCA;Pool=1\tGT\t1\r\n",
      "MN908947.3\t29402\t.\tG\tT\t2274.4\tPASS\tTotalReads=205;SupportFraction=0.989716;SupportFractionByStrand=0.99194,0.986241;BaseCalledReadsWithVariant=141;BaseCalledFraction=0.684466;AlleleCount=1;StrandSupport=1,1,124,79;StrandFisherTest=0;SOR=1.21207;RefContext=AGGCTGATGAA;Pool=1\tGT\t1\r\n",
      "MN908947.3\t29743\t.\tC\tT\t4200.2\tPASS\tTotalReads=484;SupportFraction=0.952445;SupportFractionByStrand=0.976479,0.89398;BaseCalledReadsWithVariant=391;BaseCalledFraction=0.804527;AlleleCount=1;StrandSupport=8,15,335,126;StrandFisherTest=30;SOR=1.98984;RefContext=CCACGCGGAGT;Pool=2\tGT\t1\r\n"
     ]
    }
   ],
   "source": [
    "! cat /Users/juanbolanos/Desktop/Nanopore_Pipeline/00_Compare_Variants/01_Consensus_Sequences/consensus_VCF/UNCC-6.pass.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "psychological-fighter",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_045512v2\t1058\t1059\tC1059T\r\n",
      "NC_045512v2\t14407\t14408\tC14408T\r\n",
      "NC_045512v2\t17013\t17014\tG17014T\r\n",
      "NC_045512v2\t21599\t21600\tG21600T\r\n",
      "NC_045512v2\t22017\t22018\tG22018T\r\n",
      "NC_045512v2\t22916\t22917\tT22917G\r\n",
      "NC_045512v2\t23402\t23403\tA23403G\r\n",
      "NC_045512v2\t25562\t25563\tG25563T\r\n",
      "NC_045512v2\t28886\t28887\tC28887T\r\n"
     ]
    }
   ],
   "source": [
    "! bedtools intersect -a /users/juanbolanos/Desktop/bed_to_vcf/DNA_bed_to_vcf/bed/B.1.429.bed  -b /users/juanbolanos/Desktop/bed_to_vcf/AA_bed_to_vcf/bed/B.1.429_AAmuts.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "covered-table",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B.1.429.bed B_1_351.bed B_1_429.bed P.1.bed     P_1.bed\r\n"
     ]
    }
   ],
   "source": [
    "! ls /users/juanbolanos/Desktop/bed_to_vcf/DNA_bed_to_vcf/bed/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "recent-screening",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B.1.1.7_AAmuts.bed B.1.351_AAmuts.bed B.1.429_AAmuts.bed P.1_AAmuts.bed\r\n"
     ]
    }
   ],
   "source": [
    "! ls /users/juanbolanos/Desktop/bed_to_vcf/AA_bed_to_vcf/bed/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "conscious-voltage",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
